src/web/src/app/request-watcher/service.ts

```diff
@@ -0,0 +1,71 @@
+import { Injectable, TemplateRef, ViewContainerRef } from '@angular/core';
+import { Overlay, OverlayConfig, OverlayRef, PositionStrategy } from '@angular/cdk/overlay';
+import { TemplatePortal } from '@angular/cdk/portal';
+import { Observable, throwError } from 'rxjs';
+import { tap, catchError } from 'rxjs/operators';
+
+@Injectable()
+export class RequestWatcherService {
+	public displayProgressSpinner = false;
+	appSpinner: TemplateRef<any>;
+	private progressSpinnerOverlayConfig: OverlayConfig;
+	private overlayRef: OverlayRef;
+	private vcRef: ViewContainerRef;
+
+	constructor(
+		private overlay: Overlay, 
+	) { }
+
+	createOverlay(config: OverlayConfig): OverlayRef {
+		return this.overlay.create(config);
+	}
+
+	positionGloballyCenter(): PositionStrategy {
+		return this.overlay.position()
+			.global()
+			.centerHorizontally()
+			.centerVertically();
+	}
+
+	setProgressSpinner(appSpinner: TemplateRef<any>, vcRef: ViewContainerRef) {
```

Comment:
Давай публичные методы называть с большой буквы.
```suggestion
	SetProgressSpinner(appSpinner: TemplateRef<any>, vcRef: ViewContainerRef) {
```

Reply:
Готово.